### ISSUE TITLE
load_data problem with BrainVision

### ISSUE THREAD

**Issue by teonlamont** [mne-tools/mne-python#2507]
Because the brainvision module wasn't modified for the new  `_read_segment`, and `_read_segment_file` methods, it errors with withe `raw.load_data` method.

To reproduce the error:

```python
from mne.io.brainvision.tests import test_brainvision
from mne.io import read_raw_brainvision


raw = read_raw_brainvision(test_brainvision.vhdr_path, preload=False)
raw.load_data()
```

**Comment by teonlamont** [comment]
thanks for h/t on the bug @reddigari

**Comment by Eric89GXL** [comment]
I can probably fix this on Monday unless there is someone more familiar with the Brainvision code who wants to tackle it.

**Comment by teonlamont** [comment]
might be able to do it tomorrow. what is the general way the two methods work together. I might as well fix the bci module I'm adding to handle this properly.

**Comment by Eric89GXL** [comment]
All you should need to do is provide the `_read_raw_segment` function. See how it's done in `fiff/raw.py`, basically given a set of time indices you need to return the right chunk of data.

**Comment by Eric89GXL** [comment]
(check `base.py` as well, if this isn't made clear in the base class there, then please improve the docs so the next person who wants to do this can do it more easily)

**Comment by Eric89GXL** [comment]
Fixed by #2517
